### PR TITLE
Add initial docker docs

### PR DESCRIPTION
**Pull request by tianon**
See https://github.com/docker-library/official-images/pull/987

### CODE REVIEW

File: docker/content.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+[![asciicast](https://asciinema.org/a/24707.png)](https://asciinema.org/a/24707)
+
+## start a daemon instance
+
+```console
+$ docker run --privileged --name some-docker -d docker:1.8-dind
+```
+
+This image includes `EXPOSE 2375` (the Docker port), so standard container linking will make it automatically available to the linked containers (as the following examples illustrate).
```
Comment (yosifkit) [inline docker/content.md]: Needs a link to `privileged` docs: https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration. And probably a warning to let users know that it is required for docker in docker and the dangers of privileged mode.

Comment (tianon) [inline docker/content.md]: You caught me...  Will add a note.

File: docker/content.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+```console
+$ docker run --privileged --name some-devicemapper-docker -d docker:dind --storage-driver=devicemapper
+```
+
+## Where to Store Data
+
+Important note: There are several ways to store data used by applications that run in Docker containers. We encourage users of the `%%REPO%%` images to familiarize themselves with the options available, including:
+
+-	Let Docker manage the storage of your database data [by writing the database files to disk on the host system using its own internal volume management](https://docs.docker.com/userguide/dockervolumes/#adding-a-data-volume). This is the default and is easy and fairly transparent to the user. The downside is that the files may be hard to locate for tools and applications that run directly on the host system, i.e. outside containers.
```
Comment (yosifkit) [inline docker/content.md]: database data??

File: docker/content.md (hunk 1 above)
Comment (tianon) [inline docker/content.md]: > **Note:** `--privileged` is required for Docker-in-Docker to function properly, but it should be used with care as it provides full access to the host environment, as explained [in the relevant section of the Docker documentation](https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration).

File: docker/content.md (hunk 2 above)
Comment (tianon) [inline docker/content.md]: ssshhh

Comment (tianon) [inline docker/content.md]:
```diff
--	Let Docker manage the storage of your database data [by writing the database files to disk on the host system using its own internal volume management](https://docs.docker.com/userguide/dockervolumes/#adding-a-data-volume). This is the default and is easy and fairly transparent to the user. The downside is that the files may be hard to locate for tools and applications that run directly on the host system, i.e. outside containers.
--	Create a data directory on the host system (outside the container) and [mount this to a directory visible from inside the container](https://docs.docker.com/userguide/dockervolumes/#mount-a-host-directory-as-a-data-volume). This places the database files in a known location on the host system, and makes it easy for tools and applications on the host system to access the files. The downside is that the user needs to make sure that the directory exists, and that e.g. directory permissions and other security mechanisms on the host system are set up correctly.
+-	Let Docker manage the storage of your data [by writing to disk on the host system using its own internal volume management](https://docs.docker.com/userguide/dockervolumes/#adding-a-data-volume). This is the default and is easy and fairly transparent to the user. The downside is that the files may be hard to locate for tools and applications that run directly on the host system, i.e. outside containers.
+-	Create a data directory on the host system (outside the container) and [mount this to a directory visible from inside the container](https://docs.docker.com/userguide/dockervolumes/#mount-a-host-directory-as-a-data-volume). This places the files in a known location on the host system, and makes it easy for tools and applications on the host system to access the files. The downside is that the user needs to make sure that the directory exists, and that e.g. directory permissions and other security mechanisms on the host system are set up correctly.
```

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**